src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.IO;
+using System.Linq;
+using NodeDocumentationMarkdownGenerator.Verbs;
+
+namespace NodeDocumentationMarkdownGenerator.Commands
+{
+    internal static class RenameCommand
+    {
+        internal static void HandleRename(RenameOptions opts)
+        {
+            if (opts.InputMdFile is null && opts.InputMdDirectory != null)
+            {
+                RenameDirectory(opts.InputMdDirectory, opts.MaxLength);
+            }
+            else if (opts.InputMdFile != null && opts.InputMdDirectory is null)
+            {
+                RenameFile(opts.InputMdFile);
+            }
+            else
+            {
+                Console.WriteLine("Invalid options: You can rename a single file using the file option\nor renaming multiple files in a directory (if they are longer than max length)\nusing the directory option");
+            }
+        }
+
+        private static void RenameFile(string file)
+        {
+            var extension = Path.GetExtension(file);
+            if (!extension.Equals(".md", StringComparison.InvariantCultureIgnoreCase))
+            {
+                Console.WriteLine($"Can only rename MD files: {file}");
+                return;
+            }
+
+            if (!File.Exists(file))
+            {
+                Console.WriteLine($"File not found: {file}");
+                return;
+            }
+
+            var baseName = Path.GetFileNameWithoutExtension(file);
+            var shortName = Dynamo.Utilities.Hash.GetHashFilenameFromString(baseName);
+
+            RenameFile(file, baseName, shortName);
+        }
+
+        private static void RenameFile(string file, string baseName, string shortName)
+        {
+            var content = File.ReadAllText(file);
+            content = content.Replace(baseName, shortName);
+            var path = Path.GetDirectoryName(file);
+            var newFile = Path.Combine(path, shortName + ".md");
+            File.WriteAllText(newFile, $"<!--- {baseName} --->\n" + content);
+            File.Delete(file);
```

Comment:
I wonder if we need to expect permission issues? only devs will run this command ?

Reply:
We will get exceptions if that is the case and those are catched at reported at a higher level in the handle function for the rename command. Yes, this is an internal tool. It is not shipped.